src/Server.zig

```diff
@@ -435,12 +448,21 @@ fn initializeHandler(server: *Server, _: std.mem.Allocator, request: types.Initi
         }
     }
 
-    log.info("zls initializing", .{});
     log.info("{}", .{server.client_capabilities});
-    log.info("Using offset encoding: {s}", .{@tagName(server.offset_encoding)});
+    log.info("offset encoding: {s}", .{@tagName(server.offset_encoding)});
+
+    server.updateConfiguration(.{}) catch |err| {
```

Comment:
It would seem that the `.{}` param here invalidates some of the `zls.json` settings?
Causes `    try server.resolveConfiguration(config_arena, &new_cfg);` to reevaluate the corresponding `?const []u8`s as if they haven't been set. 
```
[Trace - 2:49:05 PM] Sending request 'initialize - (0)'.
info : ( main ): Enabled debug logging.
info : ( main ): Starting ZLS 0.12.0-dev.17+4c3236b @ '/home/rad/lab/zls/npdm/zig-out/bin/zls'
info : (server): set config option 'semantic_tokens' to 'Config.Config__enum_5911.none' // Custom
info : (server): set config option 'warn_style' to 'true' // Custom
info : (server): set config option 'highlight_global_var_declarations' to 'true' // Custom
info : (server): set config option 'skip_std_references' to 'true' // Custom
info : (server): set config option 'prefer_ast_check_as_child_process' to 'false' // Custom
info : (server): set config option 'builtin_path' to '/home/rad/.cache/zls/builtin.zig' // Auto
info : (server): set config option 'zig_lib_path' to '/home/rad/lab/zig/build/stage3/lib/zig' // Auto
info : (server): set config option 'zig_exe_path' to '/home/rad/lab/zig/build/stage3/bin/zig' // Custom
info : (server): set config option 'build_runner_path' to '/home/rad/.cache/zls/build_runner.zig' // Auto
info : (server): set config option 'global_cache_path' to '/home/rad/.cache/zls' // Auto
info : (server): set config option 'build_runner_global_cache_path' to '/home/rad/.cache/zig' // Auto
info : (server): client is 'VSCodium-1.81.0'
info : (server): Server.ClientCapabilities{ .supports_snippets = true, .supports_apply_edits = true, .supports_will_save = true, .supports_will_save_wait_until = true, .supports_publish_diagnostics = true, .supports_code_action_fixall = false, .hover_supports_md = true, .completion_doc_supports_md = true, .label_details_support = true, .supports_configuration = true, .supports_workspace_did_change_configuration_dynamic_registration = true, .supports_textDocument_definition_linkSupport = true }
info : (server): offset encoding: utf-16
info : (server): set config option 'zig_lib_path' to '/home/rad/lab/zig/lib' // Auto
info : (server): set config option 'zig_exe_path' to '/home/rad/lab/zig/build/stage4/bin/zig' // Auto from PATH
```

Reply:
does #1404 resolve your issue?